packages/open-vm-tools/0003-Update-shutdown-code-to-work-for-Bottlerocket.patch

```diff
@@ -0,0 +1,106 @@
+From b948c8ed5edafe0683b17f4808dfdf5de2c1737c Mon Sep 17 00:00:00 2001
+From: Matthew Yeazel <yeazelm@amazon.com>
+Date: Fri, 2 Jun 2023 16:18:12 -0700
+Subject: [PATCH] Update shutdown code to work for Bottlerocket
+
+Bottlerocket doesn't ship with a shell so system() doesn't function
+correctly. This commit replaces the rest of this logic to specifically
+work for Bottlerocket and only Bottlerocket. A separate commit will be
+made to attempt to add support for Bottlerocket alongside the other
+supported operating systems.
+
+Signed-off-by: Matthew Yeazel <yeazelm@amazon.com>
+---
+ lib/include/system.h     |  1 +
+ lib/system/systemLinux.c | 62 ++++++++++++++++----------
+ 2 files changed, 39 insertions(+), 24 deletions(-)
+
+diff --git a/lib/include/system.h b/lib/include/system.h
+index f1119cef..b82b2a26 100644
+--- a/lib/include/system.h
++++ b/lib/include/system.h
+@@ -37,6 +37,7 @@
+ uint64 System_GetTimeMonotonic(void);
+ uint64 System_Uptime(void);
+ void System_Shutdown(Bool reboot);
++void Bottlerocket_Shutdown(Bool reboot);
```

Comment:
Would it make sense to avoid calling this function `Bottlerocket_Shutdown`, and rework the patch to check if `/bin/sh` exists, and if it does call `system`? With the hope that probably the VMware folks will take this patch upstream? :crossed_fingers:

Reply:
@bcressey helped me find a better option which is `g_spawn_command_line_sync()` which should be a more upstream-able change.

Reply:
@arnaldo2792: If in doubt, you can also invoke `system(NULL)` to determine whether prerequisites are met.

TIL about `g_spawn_command_line_sync`. Nifty! Luckily there's a dependency on GLib already.